### ISSUE TITLE
version 1.2 & 1.2.1 no longer download plugin dependencies

### ISSUE THREAD

**Issue by leoherbie** [qaware/go-offline-maven-plugin#12]
I have the maven enforcer plugin configured like the following.  Note the extra dependencies included:

```
<profile>
	<id>enforcer</id>
	<properties>
		<extra-enforcer-rules.version>1.2</extra-enforcer-rules.version>
		<maven-enforcer-plugin.version>3.0.0-M2</maven-enforcer-plugin.version>
		<ossindex-maven-enforcer-rules.version>3.0.4</ossindex-maven-enforcer-rules.version>
		<enforcer.phase>validate</enforcer.phase>
	</properties>
	<build>
		<pluginManagement>
			<plugins>
				<plugin>
					<groupId>org.apache.maven.plugins</groupId>
					<artifactId>maven-enforcer-plugin</artifactId>
					<version>${maven-enforcer-plugin.version}</version>
					<executions>
						<execution>
							<id>enforcer-rules</id>
							<goals>
								<goal>enforce</goal>
							</goals>
							<phase>${enforcer.phase}</phase>
							<configuration>
								<rules>
									<banVulnerable implementation="org.sonatype.ossindex.maven.enforcer.BanVulnerableDependencies">
									</banVulnerable>
							</configuration>
						</execution>
					</executions>
					<dependencies>
						<dependency>
							<groupId>org.codehaus.mojo</groupId>
							<artifactId>extra-enforcer-rules</artifactId>
							<version>${extra-enforcer-rules.version}</version>
						</dependency>
						<dependency>
							<groupId>org.sonatype.ossindex.maven</groupId>
							<artifactId>ossindex-maven-enforcer-rules</artifactId>
							<version>${ossindex-maven-enforcer-rules.version}</version>
						</dependency>
					</dependencies>
				</plugin>
			</plugins>
		</pluginManagement>
		<plugins>
			<plugin>
				<groupId>org.apache.maven.plugins</groupId>
				<artifactId>maven-enforcer-plugin</artifactId>
			</plugin>
		</plugins>
	</build>
</profile>
```

When I activate the profile, the go-offline plugin version 1.0 downloads the maven-enforcer-plugin along with it's dependencies: extra-enforcer-rules & ossindex-maven-enforcer-rules.  With version 2.0 & 2.1, the go-offline plugin no downloads the plugin's dependencies.

**Comment by Theoderich** [comment]
Thanks for the report, I will look into it

**Comment by Theoderich** [comment]
Just tried it with the minimal pom.xml below.

Calling
`mvn de.qaware.maven:go-offline-maven-plugin:1.2.1:resolve-dependencies -P enforcer`
does download the enforcer-plugin, extra-enforcer-rules and ossindex-maven-enforcer-rules for me.
Can you test if it works for you with the minimal pom? What Maven version are you using?


```xml
<?xml version="1.0" encoding="UTF-8"?>
<project xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
         xmlns="http://maven.apache.org/POM/4.0.0"
         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
    <modelVersion>4.0.0</modelVersion>

    <groupId>de.qaware.maven</groupId>
    <artifactId>enforcer-test</artifactId>
    <version>1.0-SNAPSHOT</version>
    <packaging>pom</packaging>

    <profiles>
        <profile>
            <id>enforcer</id>
            <properties>
                <extra-enforcer-rules.version>1.2</extra-enforcer-rules.version>
                <maven-enforcer-plugin.version>3.0.0-M2</maven-enforcer-plugin.version>
                <ossindex-maven-enforcer-rules.version>3.0.4</ossindex-maven-enforcer-rules.version>
                <enforcer.phase>validate</enforcer.phase>
            </properties>
            <build>
                <pluginManagement>
                    <plugins>
                        <plugin>
                            <groupId>org.apache.maven.plugins</groupId>
                            <artifactId>maven-enforcer-plugin</artifactId>
                            <version>${maven-enforcer-plugin.version}</version>
                            <executions>
                                <execution>
                                    <id>enforcer-rules</id>
                                    <goals>
                                        <goal>enforce</goal>
                                    </goals>
                                    <phase>${enforcer.phase}</phase>
                                    <configuration>
                                        <rules>
                                            <banVulnerable
                                                    implementation="org.sonatype.ossindex.maven.enforcer.BanVulnerableDependencies">
                                            </banVulnerable>
                                        </rules>
                                    </configuration>
                                </execution>
                            </executions>
                            <dependencies>
                                <dependency>
                                    <groupId>org.codehaus.mojo</groupId>
                                    <artifactId>extra-enforcer-rules</artifactId>
                                    <version>${extra-enforcer-rules.version}</version>
                                </dependency>
                                <dependency>
                                    <groupId>org.sonatype.ossindex.maven</groupId>
                                    <artifactId>ossindex-maven-enforcer-rules</artifactId>
                                    <version>${ossindex-maven-enforcer-rules.version}</version>
                                </dependency>
                            </dependencies>
                        </plugin>
                    </plugins>
                </pluginManagement>
                <plugins>
                    <plugin>
                        <groupId>org.apache.maven.plugins</groupId>
                        <artifactId>maven-enforcer-plugin</artifactId>
                    </plugin>
                </plugins>
            </build>
        </profile>
    </profiles>
</project>
```

**Comment by Theoderich** [comment]
> Sure. I'll gladly test for you. Thanks for looking into this. I'm not sure what you mean by "minimal" pom. Just pull it out of a profile or something?

I meant using the pom.xml I posted in my previous comment

**Comment by leoherbie** [comment]
oh gee.  I didn't notice that was a whole pom.  ok.  I'll find some time to test today.

**Comment by Theoderich** [comment]
Any updates on this problem?

**Comment by Theoderich** [comment]
Any updates here?

**Comment by Theoderich** [comment]
Should be fixed with 1.2.3
